File: agentauth-sdk-python/examples/basic_usage.py
"""
Basic usage example for AgentAuth Python SDK
"""

import asyncio
from agentauth_sdk import AgentAuthClient, Permissions


async def main():
    print("🚀 AgentAuth Python SDK Example\n")

    # Initialize client
    async with AgentAuthClient(base_url="http://localhost:3000") as client:
        # 1. Register an agent
        print("1️⃣ Registering agent...")
        result = await client.register_agent(
            name="Example Support Agent",
            owner_email="example@company.com",
            permissions=[
                Permissions.Zendesk.Tickets.Read,
                Permissions.Zendesk.Tickets.Write,
                Permissions.Slack.Messages.Write,
                Permissions.HubSpot.Contacts.Read,
            ],
        )

        print(f"✅ Agent registered: {result.agent.agent_id}")
        print(f"   API Key: {result.credentials.api_key[:20]}...")
        print(f"   Permissions: {len(result.agent.permissions)}")

        # 2. Verify agent
        print("\n2️⃣ Verifying agent...")
        verify_result = await client.verify_agent(
            agent_id=result.agent.agent_id,
            api_key=result.credentials.api_key,
        )

        print("✅ Agent verified")
        print(f"   Access Token: {verify_result.token.access_token[:30]}...")
        print(f"   Expires in: {verify_result.token.expires_in}s")

        # 3. Get agent details
        print("\n3️⃣ Fetching agent details...")
        agent = await client.get_agent(result.agent.agent_id)

        print("✅ Agent details retrieved:")
        print(f"   Name: {agent.name}")
        print(f"   Status: {agent.status}")
        print(f"   Tier: {agent.tier}")

        # 4. Get activity logs
        print("\n4️⃣ Fetching activity logs...")
        activity = await client.get_activity(
            result.agent.agent_id,
            limit=10,
        )

        print("✅ Activity logs retrieved:")
        print(f"   Total events: {activity.pagination.total}")
        print(f"   Recent events: {len(activity.activity)}")

        # 5. List permissions
        print("\n5️⃣ Listing available permissions...")
        permissions = await client.list_permissions()

        print("✅ Available permissions:")
        for service, perms in permissions["permissions"].items():
            print(f"   {service}: {len(perms)} permissions")

        # 6. Health check
        print("\n6️⃣ Checking API health...")
        health = await client.health_check()

        print("✅ Health check:")
        print(f"   Status: {health.status}")
        print(f"   Database: {health.database}")

        print("\n🎉 Example completed successfully!")


if __name__ == "__main__":
    asyncio.run(main())

File: agentauth-sdk/examples/basic-usage.ts
/**
 * Basic usage example for AgentAuth SDK
 *
 * Run with: npx tsx examples/basic-usage.ts
 */

import { AgentAuthClient, Permissions, type Permission } from '../src';

async function main() {
  // Initialize client
  const client = new AgentAuthClient({
    baseURL: process.env.AGENTAUTH_URL || 'http://localhost:3000',
  });

  console.log('🚀 AgentAuth SDK Example\n');

  // 1. Register an agent
  console.log('1️⃣ Registering agent...');
  const permissions: Permission[] = [
    Permissions.Zendesk.Tickets.Read,
    Permissions.Zendesk.Tickets.Write,
    Permissions.Slack.Messages.Write,
    Permissions.HubSpot.Contacts.Read,
  ];

  const { agent, credentials } = await client.registerAgent({
    name: 'Example Support Agent',
    owner_email: 'example@company.com',
    permissions,
  });

  console.log('✅ Agent registered:', agent.agent_id);
  console.log('   API Key:', credentials.api_key.substring(0, 20) + '...');
  console.log('   Permissions:', agent.permissions.length);

  // 2. Verify agent and get JWT token
  console.log('\n2️⃣ Verifying agent...');
  const { token } = await client.verifyAgent({
    agent_id: agent.agent_id,
    api_key: credentials.api_key,
  });

  console.log('✅ Agent verified');
  console.log('   Access Token:', token.access_token.substring(0, 30) + '...');
  console.log('   Expires in:', token.expires_in, 'seconds');

  // 3. Get agent details
  console.log('\n3️⃣ Fetching agent details...');
  const agentDetails = await client.getAgent(agent.agent_id);

  console.log('✅ Agent details retrieved:');
  console.log('   Name:', agentDetails.agent.name);
  console.log('   Status:', agentDetails.agent.status);
  console.log('   Tier:', agentDetails.agent.tier);
  console.log('   Created:', new Date(agentDetails.agent.created_at).toLocaleString());

  // 4. Get activity logs
  console.log('\n4️⃣ Fetching activity logs...');
  const activity = await client.getActivity(agent.agent_id, { limit: 10 });

  console.log('✅ Activity logs retrieved:');
  console.log('   Total events:', activity.pagination.total);
  console.log('   Recent events:', activity.activity.length);

  // 5. List all available permissions
  console.log('\n5️⃣ Listing available permissions...');
  const permissionsList = await client.listPermissions();

  console.log('✅ Available permissions:');
  Object.entries(permissionsList.permissions).forEach(([service, perms]) => {
    console.log(`   ${service}: ${perms.length} permissions`);
  });

  // 6. Health check
  console.log('\n6️⃣ Checking API health...');
  const health = await client.healthCheck();

  console.log('✅ Health check:');
  console.log('   Status:', health.status);
  console.log('   Database:', health.database);

  console.log('\n🎉 Example completed successfully!');
}

// Run the example
main().catch((error) => {
  console.error('❌ Error:', error.message);
  if (error.body) {
    console.error('   Details:', error.body);
  }
  process.exit(1);
});
